### PR TITLE
refactor: replace dynamic sql in getUserInfo

### PR DESCRIPTION
**Pull request by Jealee3000**
Submit a pull request for this project.

# Why? 

refactor: multiple joins query is prohibited #2848

# What?

use several small queries instead of multiple joins query

# How?

- [x] add `selectUserBaseInfoByUuids`  func
- [x] add a parameter `unitType` to `getMembersBaseInfoBySpaceIdAndUserIds` func
- [x] modify unit test

### CODE REVIEW

File: packages/room-server/src/user/repositories/user.repository.ts [hunk 1]
```diff
@@ -76,6 +77,18 @@ export class UserRepository extends Repository<UserEntity> {
...
   }
 
+  /**
+   * Query user info by user uuid array
+   */
+  public async selectUserBaseInfoByUuids(uuids: string[]): Promise<UserBaseInfoDto[]> {
+    return await this.find({
+      select: ['id', 'uuid', 'avatar', 'nikeName', 'color', 'isSocialNameModified'],
+      where: [{
+        uuid: In(uuids),
```
Comment (ChambersChan) [inline packages/room-server/src/user/repositories/user.repository.ts:87]: add default condition `isDeleted: false`.

Comment (Jealee3000) [reply packages/room-server/src/user/repositories/user.repository.ts:87]: > add default condition `isDeleted: false`.

no `isDeleted` query in the `selectUserInfoBySpaceIdAndUuids` func.
even so, should I add the default condition `isDeleted: false` ?

Comment (ChambersChan) [reply packages/room-server/src/user/repositories/user.repository.ts:87]: Yes. The lack of condition in origin method is our mistake.

**Review by ChambersChan — APPROVED** [review]

**Comment by yort-feng** [comment]
This PR led to many BUGs, we decided to revert it first.

**Comment by Jealee3000** [comment]
> This PR led to many BUGs, we decided to revert it first.

Sorry about this, can you tell me more about BUGs?

**Pull request merged**